census/vanilla-tf/trainer/task.py

```diff
@@ -44,228 +43,149 @@
+  """Run the training steps and calculate accuracy every 10 steps."""
 
-  if job_name == 'ps':
-    server.join()
-  elif job_name in ['master', 'worker']:
+  training_eval_graph = tf.Graph()
+  with training_eval_graph.as_default():
     with tf.device(tf.train.replica_device_setter()):
-      inputs, labels = get_placeholders()
-      graph = make_graph(inputs, labels)
-
-    init = tf.global_variables_initializer()
-
-    with tf.train.MonitoredTrainingSession(master=server.target,
+      mode = tf.placeholder(shape=[], dtype=tf.string)
+      eval_features, eval_label = input_fn(
+          eval_data_paths, shuffle=False, batch_size=batch_size)
+      train_features, train_label = input_fn(
+          train_data_paths, shuffle=True, num_epochs=num_epochs, batch_size=batch_size)
+
+      is_train = tf.equal(mode, tf.constant(TRAIN))
+      sorted_keys = train_features.keys()
+      sorted_keys.sort()
+      inputs = dict(zip(
+          sorted_keys,
+          tf.cond(
+              is_train,
+              lambda: [train_features[k] for k in sorted_keys],
+              lambda: [eval_features[k] for k in sorted_keys]
+          )
+      ))
+      labels = tf.cond(is_train, lambda: train_label, lambda: eval_label)
+      train_op, accuracy_op, eval_op, global_step_tensor, predictions = model_fn(
+          inputs, labels, learning_rate=learning_rate)
+
+    with tf.train.MonitoredTrainingSession(master=target,
                                            is_chief=is_chief,
                                            checkpoint_dir=job_dir,
                                            save_checkpoint_secs=20,
                                            save_summaries_steps=50) as session:
-      session.run(init)
-      training_steps(session, graph, inputs, labels, max_steps,
-                     train_eval_tensor, job_name=job_name, job_id=task_index)
-
-
-def evaluate_accuracy(session, eval_step, inputs, labels,
-                      eval_input, eval_label):
-  """Perform the evaluation step to calculate accuracy."""
-  return 100 * session.run(
-      eval_step,
-      feed_dict={
-          inputs: session.run(eval_input),
-          labels: session.run(eval_label)
-      }
-  )
-
-
-def parse_tf_config():
+      coord = tf.train.Coordinator()
+      tf.train.start_queue_runners(coord=coord, sess=session)
+      step = 0
+      last_eval = 0 
+      with coord.stop_on_exception():
+        while step < max_steps and not coord.should_stop():
+            if is_chief and step - last_eval > eval_every:
+                last_eval = step
+                for _ in range(eval_steps):
+                    session.run([eval_op], feed_dict={mode: EVAL})
+                accuracy = session.run([accuracy_op], feed_dict={mode: EVAL})
+                print("Accuracy at step: {} is {}".format(step, accuracy))
+            step, _ = session.run(
+                [global_step_tensor, train_op],
+                feed_dict={mode: TRAIN}
```

Comment:
@puneith No it wouldn't be addressed by distributed. What you're proposing only works if you have a dedicated evaluation master. Otherwise you need a way to have continuous evaluation in one task. This requires either loading from checkpoints (tying to check-pointing) or merging the two graphs.